### PR TITLE
CLN: simplify _shallow_copy

### PR DESCRIPTION
**Pull request by jbrockmendel**
The class hierarchy of in/around groupby is really tough, this simplifies it a little bit

### CODE REVIEW

**Comment by WillAyd** [comment]
Is `GroupByMixin` used at all within groupby? Maybe worth moving the entire class

**Comment by jbrockmendel** [comment]
huh, looks like it isnt.  any strong opinion on where it would be moved?

**Comment by WillAyd** [comment]
Maybe merge into WindowMixin. If it’s not actually used for GroupBy I don’t even think we need to keep the GroupByMixin name as that would be rather misleading

**Comment by jreback** [comment]
let me look 

I made these for consistency between group by and window ops

**Comment by jbrockmendel** [comment]
> Maybe merge into WindowMixin. If it’s not actually used for GroupBy I don’t even think we need to keep the GroupByMixin name as that would be rather misleading

For now at least I'm leaning towards keeping it where it is.  it would need to go into some shared-by-resample/window location, which describes a lot of groupby code anyway.

File: pandas/core/window/common.py [hunk 1]
```diff
@@ -26,7 +26,23 @@
 """
 
 
-class _GroupByMixin(GroupByMixin):
+def _dispatch(name: str, *args, **kwargs):
```
Comment (jreback) [inline pandas/core/window/common.py]: eventually add a full doc-string if we end up using this

**Comment by jreback** [comment]
thxs

**Pull request merged**